### PR TITLE
Help users more to set configuration for later use in CI

### PR DESCRIPTION
**Pull request by dpoirier**
As long as a user has kubectl set up to access their kubernetes cluster, they can start using this role and it'll set up a deploy account (if there's not one already) and print out instructions to configure ansible to use it.

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -3,9 +3,6 @@
 # ansible-role-django-k8s performs a two-step deploy process.
 # First, it uses your (admin) credentials to create a namespace and service account;
 # those admin credentials should be defined with a combination of these variables:
-# (i.e., in your local ~/.kube/config):
-k8s_kube_config: "~/.kube/config"
-k8s_kube_context: ""
```
Comment (copelco) [inline defaults/main.yml:8]: Do these variables need to be removed? Maybe I'm missing something, but it seems too easy to accidentally run these tasks on the wrong cluster if working on multiple projects, which doesn't seem too uncommon.

File: tasks/main.yml [hunk 2]
```diff
@@ -9,53 +9,78 @@
...
+  block:
+    - command: kubectl config view --minify=true -o jsonpath='{.clusters[0].cluster.certificate-authority-data}' --raw
+      register: data_command
+    - debug:
+         msg: |
+           1) Create a file (maybe k8s_auth_ssl_ca_cert.txt) and set contents to
+           {{ data_command.stdout | b64decode }}
+           with all indentation removed. (This does not need to be secret.)
+           2) Set k8s_auth_ssl_ca_cert to the path to the file.
+           3) Run again after doing that
```
Comment (copelco) [inline tasks/main.yml:62]: 🆒

File: tasks/main.yml [hunk 3]
```diff
@@ -9,53 +9,78 @@
...
+    k8s_info:
+      kind: Secret
+      namespace: "{{ k8s_namespace }}"
+      name: "{{ deploy_account_name }}"
+    register: deploy_account_secret
+
+  - debug:
+      msg: |
+        # Set a variable as follows:
+        k8s_auth_api_key: "{{ deploy_account_secret.resources.0.data.token | b64decode }}"
```
Comment (copelco) [inline tasks/main.yml:104]: Nice!

File: defaults/main.yml (hunk 1 above)
Comment (dpoirier) [reply defaults/main.yml:8]: That's a good point. I like the idea of something in version control that would nail down which cluster we are supposed to be deploying to.

I've updated this to use ``k8s_auth_host`` for that. It'll require ``k8s_auth_host`` to have been set already, and it'll specify that as the host on all k8s tasks to make sure we're using that cluster. How does that look?

**Comment by dpoirier** [comment]
@copelco If we like this approach, we might want to apply it to https://github.com/caktus/ansible-role-k8s-web-cluster as well?

File: README.rst [hunk 4]
```diff
@@ -61,12 +63,57 @@ Installation
...
               k8s_domain_names:
               - www.example.com
 
 See ``defaults/main.yml`` for all the variables that can be overridden.
 
+The ``k8s_auth_host`` variable is absolutely required to be set. This is the API
+endpoint URL of the cluster to use. Here are some examples so you can see what
+it might look like:
+
+AWS: ``https://74406E3AD450E7845D0EF653E7C6F020.gr7.us-west-2.eks.amazonaws.com``
```
Comment (dpoirier) [inline README.rst:78]: @copelco what would this URL look like for a cluster in Azure?

File: defaults/main.yml (hunk 1 above)
Comment (copelco) [reply defaults/main.yml:8]: I think this makes sense. Since the approach is different (basically the opposite) from what #13 was suggesting, it took me a bit to wrap my head around it. I was struggling with:

* I'm currently deploying to multiple clusters quite often, which I could achieve previously with a simple ``ansible-playbook -l my-cluster`` limit and it would pick up the correct ``k8s_kube_context`` automatically. 
* Additionally, we don't currently have ``k8s_auth_api_key`` committed to the repo anywhere. In fact, one of our repos isn't using ``ansible-vault`` at all. CI uses an environment variable and I didn't need it locally using my ``k8s_kube_context``.

So at first I felt like requiring the user to have the current context active and ``k8s_auth_api_key`` set was a drawback, but I see the value of requiring use of the deploy account. Perhaps #13 should just be closed.

I haven't used Digitial Ocean K8s yet, but I've noticed use of ``k8s_kube_config`` in conjunction with DO, like [here](https://github.com/caktus/caktus-hosting-services/blob/master/Ansible/services/inventory/host_vars/inddex-cluster.yaml#L4). I don't know why this is. I assume the new configuration will still work? Maybe @tobiasmcnulty or @vkurup can provide insight.

File: tasks/main.yml [hunk 5]
```diff
@@ -9,59 +9,89 @@
...
+           2) Set k8s_auth_ssl_ca_cert to the path to the file.
+           3) Run again after doing that
 
-- name: Set k8s_auth_api_key
-  set_fact:
-    k8s_auth_api_key: "{{ deploy_account_token.resources.0.data.token | b64decode }}"
-  when: not k8s_auth_api_key
+# Does the user need to set k8s_auth_api_key?
+- when: not k8s_auth_api_key
+  block:
```
Comment (copelco) [inline tasks/main.yml:66]: I kept running into a hard to parse ``SSLError`` stack trace here and wondered if adding tasks here with a helpful error message would be useful to other users:

```yaml
block:
  - name: Query nodes to check cluster connection
    k8s_info:
      kind: Node
      host: "{{ k8s_auth_host }}"
    register: node_query_result
    ignore_errors: True

  - name: Fail play if above Node query was unsuccessful
    fail:
      msg: "Unable to access the cluster. Are you using the correct kubectl context?"
    when:
      - "'rc' in node_query_result"
      - node_query_result.rc == 1
```

File: README.rst (hunk 4 above)
Comment (copelco) [reply README.rst:78]: Something like this:
```suggestion
AKS: ``https://ratom-staging-dns-ba5d6fd2.hcp.eastus.azmk8s.io:443``

AWS: ``https://74406E3AD450E7845D0EF653E7C6F020.gr7.us-west-2.eks.amazonaws.com``
```

**Review by copelco — APPROVED** [review]
Added some comments and responded to others.

We should include in the release notes that this will break previous versions and will require changes since ``k8s_auth_api_key`` is now required.

**Comment by dpoirier** [comment]
@copelco I really like the way we have another project set up so if CI passes on certain branches the CI deploys it. Deploying keeps getting more complicated now that we have to also build a docker image and put it in the right place and tag it, so that simplifies things once it is set up. So I've ended up wanting to support the flow where one developer gets things set up with a deploy account, and then things happen automatically without devs all having to know the details of how to deploy.

File: tasks/main.yml (hunk 5 above)
Comment (dpoirier) [reply tasks/main.yml:66]: This seems like a good idea. I think I ran into a similar error and meant to at least add a note to the docs, but never did.  I'll add something like this.

**Comment by dpoirier** [comment]
@copelco I updated the CHANGES, and wanted to make it simple to get people's definitions updated so I ended up refactoring the tasks a bit so anytime someone was running it with kubectl access working, we'd take the opportunity to update the deploy account's definition. I'd appreciate if you could take another look before I merge this (and call it 0.0.2, I think).

**Comment by dpoirier** [comment]
@copelco have you had a chance to look at the most recent changes on this? (after your approval)

**Review by copelco — APPROVED** [review]
Just tested with @JeremyGibson on a new project. Looks good and the help messages are super useful! 🚢
